eos_data_distribution/avahi_monitor.py

```diff
@@ -79,23 +96,102 @@ def __init__(self):
         check_call(
             ["nfdc", "set-strategy", str(SUBSCRIPTIONS_SOMA), 'ndn:/localhost/nfd/strategy/multicast'])
 
-        self._registry = dict()
+        self.update_routed_status()
+        self.process_route_changes(False)
+
+    def update_routed_status(self):
+        self._routed = set(
+            self._ndn_gateways.values()).intersection(self.gateways)
+
+    def process_route_changes(self, prev_routed):
+        if self._routed:
+            logger.info(
+                "Being routed by an NDN node at %s (via %s), NOT acting as an edge router", address, ifname)
+            if not prev_routed:
+                # need to remove old links...
+                self._registery.values().map(self.remove_nexthop)
+                # ..and only add one to the routers
+                self._ndn_gateways.keys().map(self.add_nexthop)
+            else:
+                # no change: i am routed, i was routed before: no links
+                pass
+        else:
+            logger.info(
+                "Not Being routed by an NDN node, acting as an edge router")
+            if prev_routed:
+                # need to add back old links...
+                # this is limited to MAX_PEERS in add_nexthop
+                self._registry.values().map(self.add_nexthop)
```

Comment:
actually i do, but in removed_cb, because here if my disappeared host was a gateway it's not in `_ndn_gateways` anymore.
the other `_ndn_gateways` are either still there (but don't route us anymore) or we'll get called again to remove them.